File: OOPython/Inheritance/TechProfessional.py

# Example of Abstract Class with abstract and virtual methods

class TechProfessional:
   
    def __init__(self,first_name,last_name):
        self.first_name = first_name
        self.last_name = last_name

    def FullName(self):
        return "{0} {1}".format(self.first_name,self.last_name)

    def Position():
        pass

    def Experience(self):
        print("My name is {0}. I am an IT General professional with over 10 years of experience!".format(self.FullName()))

File: OOPcsharp/Inheritance/TechProfessional.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace OOPcsharp.Inheritance
{
    /*
     * Example of Abstract Class with abstract and virtual methods
     * 
     * */

    public abstract class TechProfessional
    {
        private string _firstName;

        private string _lastName;

        public TechProfessional(string firstName, string lastName)
        {
            _firstName = firstName;
            _lastName = lastName;
        }

        public string FullName
        {
            get { return string.Format("{0} {1}", this._firstName, this._lastName); }
        }

        public abstract void Position();

        public virtual void Experience()
         {
             Console.WriteLine(string.Format("My name is {0}. I am an IT General professional with over 10 years of experience!", this.FullName));
         }

    }
}
